blueocean-dashboard/src/main/js/PipelineRoutes.jsx

```diff
@@ -75,24 +76,52 @@ function persistModalBackground() {
     root.appendChild(container);
 }
 
+function isEnteringRunDetails(prevState, nextState) {
```

Comment:
all these fiddly things look like great places for bugs to hide - unit testable?